src/PelJpeg.php

```diff
@@ -401,16 +404,18 @@ public function getICC()
     /**
      * Clear any Exif data.
      *
-     * This method will only clear the first @{link PelJpegMarker::APP1}
-     * section found (there should normally be just one).
+     * This method will only clear @{link PelJpegMarker::APP1} EXIF sections found.
      */
     public function clearExif()
     {
-        $sections_count = count($this->sections);
-        for ($i = 0; $i < $sections_count; $i ++) {
-            if ($this->sections[$i][0] == PelJpegMarker::APP1) {
-                unset($this->sections[$i]);
-                return;
+        $idx = 0;
+        while ($idx < count($this->sections)) {
+            $s = $this->sections[$idx];
+            if (($s[0] == PelJpegMarker::APP1) && ($s[1] instanceof PelExif)) {
+                array_splice($this->sections, $idx, 1);
+                $idx--;
```

Comment:
Why is $idx decremented here?

Reply:
Index has to be rewind after an item was removed, as following items' are shifted back. Reflected this in the test.

Reply:
ok